### PR TITLE
#532: Add config to FileUploaderField to customise button and allow external control of loading status

### LINKED ISSUE #532: Add config to FileUploaderField to customise button and allow external control of loading status

### PR DESCRIPTION
**Pull request by federico-ercoles**
Closes #532

## Test Plan
✅ Button kind is configurable correctly
✅ `isUploading` prop correctly forces the component in uploading status

### CODE REVIEW

File: packages/bento-design-system/src/FileUploaderField/FileUploaderField.tsx [hunk 1]
```diff
@@ -43,11 +43,6 @@ function isDropzoneKnownError(error: string): error is KnownDropzoneErrorCode {
   return errorCodes.find((c) => c === error) !== undefined;
 }
 
-export type ErrorMessages<E extends string> = Record<
-  E | ErrorCode,
-  Children | ((file: File | undefined) => Children)
->;
-
```
Comment (federico-ercoles) [inline packages/bento-design-system/src/FileUploaderField/FileUploaderField.tsx:50]: This was unused.

File: packages/storybook/stories/Components/FileUploaderField.stories.tsx [hunk 2]
```diff
@@ -21,10 +22,36 @@ export const fileUploaderField = createControlledStory(undefined, {
...
+
+export const fileUploaderField = createControlledStory(undefined, fileUploaderProps);
+
+export const loading = createControlledStory(undefined, {
+  ...fileUploaderProps,
+  isUploading: true,
 });
+loading.parameters = {
+  chromatic: { pauseAnimationAtEnd: true },
+};
```
Comment (federico-ercoles) [inline packages/storybook/stories/Components/FileUploaderField.stories.tsx:42]: I'm not sure if this is the right parameter to ensure consistency between screenshots, I've also seen used `delay` in `AreaLoader`'s stories.

File: packages/bento-design-system/src/FileUploaderField/FileUploaderField.tsx [hunk 3]
```diff
@@ -141,10 +140,11 @@ export function FileUploaderField<E extends string>({
   maxFileSize,
   allowedFileTypes,
   height: height_,
+  isUploading,
 }: Props<E>) {
   const config = useBentoConfig().fileUploaderField;
   const height = height_ ?? config.defaultHeight;
-  const [uploading, setUploading] = useState<boolean>(false);
+  const [uploading, setUploading] = useState<boolean>(isUploading ?? false);
```
Comment (veej) [inline packages/bento-design-system/src/FileUploaderField/FileUploaderField.tsx:147]: I'm not sure this is making the component fully Controlled. In this way, the `isUploading` prop will be used only to initialize the state, but a change of the prop from outside will not have effect on the internal state.

If you want to change the state on prop change, you need to manually handle it with a `useEffect`:

```typescript
useEffect(() => {
   setUploading(isUploading);
}, [isUploading])
```

**Review by veej — CHANGES_REQUESTED** [review]

File: packages/bento-design-system/src/FileUploaderField/FileUploaderField.tsx (hunk 3 above)
Comment (federico-ercoles) [reply packages/bento-design-system/src/FileUploaderField/FileUploaderField.tsx:148]: Should be correct now:

https://user-images.githubusercontent.com/26444095/221237774-d70c0a72-b520-4f26-a482-68c35cb469de.mov

**Review by veej — APPROVED** [review]
💯

**Review by gabro — APPROVED** [review]

**Pull request merged**